src/io/flutter/actions/FlutterSdkAction.java

```diff
@@ -30,19 +32,43 @@
   public void actionPerformed(AnActionEvent event) {
     final Project project = DumbAwareAction.getEventProject(event);
 
-    final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
-    if (sdk == null) {
-      showMissingSdkDialog(project);
-      return;
+    // See if the Bazel workspace exists for this project.
+    final Workspace workspace = project != null ? WorkspaceCache.getInstance(project).getNow() : null;
+
+    if (enableActionInBazelContext() && workspace != null) {
+      FlutterInitializer.sendAnalyticsAction(this);
+      FileDocumentManager.getInstance().saveAllDocuments();
+      startCommandInBazelContext(project, workspace);
     }
+    else {
+      final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
+      if (sdk == null) {
+        showMissingSdkDialog(project);
+        return;
+      }
 
-    FlutterInitializer.sendAnalyticsAction(this);
-    FileDocumentManager.getInstance().saveAllDocuments();
-    startCommand(project, sdk, PubRoot.forEventWithRefresh(event));
+      FlutterInitializer.sendAnalyticsAction(this);
+      FileDocumentManager.getInstance().saveAllDocuments();
+      startCommand(project, sdk, PubRoot.forEventWithRefresh(event));
+    }
   }
 
   public abstract void startCommand(@NotNull Project project, @NotNull FlutterSdk sdk, @Nullable PubRoot root);
 
+  /**
+   * Implemented by actions which are used in the Bazel context ({@link #enableActionInBazelContext()} returns true), by default this method
+   * is a no-op.
+   */
+  public void startCommandInBazelContext(@NotNull Project project, @NotNull Workspace workspace) {
```

Comment:
How about throwing an exception?  It seems like calling the no-op version would be a logic error?